lib/s3encrypt.rb

```diff
@@ -126,6 +126,24 @@ def self.getfile(local_filename, remote_filename, bucket, app_context)
     fetch_file(s3client,keyvalue,local_filename,remote_filename,bucket)
   end
 
+  #########################################
+  # Helper method that bypasses writing a file to the system and returns a JSON object
+  # More for accessiing the data programatically so that a file does not have to be written and cleand up
+  #########################################
+  def self.getfile_as_json(remote_filename, bucket, app_context)
+    resp = getfile(nil, remote_filename, bucket, app_context)
+    return JSON.parse(resp.body.string)
```

Comment:
You are correct. Sorry about that. My script that used this had a require
line in it from when I was testing.



On Sep 10, 2016 19:31, "DonMills" <notifications@github.com> wrote:

> In lib/s3encrypt.rb
> <https://github.com/DonMills/ruby-kms-s3-gem/pull/1#discussion_r78285648>:
>
> > @@ -126,6 +126,24 @@ def self.getfile(local_filename, remote_filename, bucket, app_context)
> >      fetch_file(s3client,keyvalue,local_filename,remote_filename,bucket)
> >    end
> >
> > +  #########################################
> > +  # Helper method that bypasses writing a file to the system and returns a JSON object
> > +  # More for accessiing the data programatically so that a file does not have to be written and cleand up
> > +  #########################################
> > +  def self.getfile_as_json(remote_filename, bucket, app_context)
> > +    resp = getfile(nil, remote_filename, bucket, app_context)
> > +    return JSON.parse(resp.body.string)
>
> Do we need to require the json module for this? It seems so but I don't
> see that in the pull request (which I already accepted)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/DonMills/ruby-kms-s3-gem/pull/1/files/768571ec9f5041b65dca27265ec372f3af3e2ac8#r78285648>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALpAwKuNxrKBbKiKC-_i1Ovor_r_fZWzks5qoz2-gaJpZM4J5gdm>
> .
>